src/connection.rs

```diff
@@ -1,3 +1,6 @@
+// There is currently a bug in clippy: https://github.com/rust-lang-nursery/rust-clippy/issues/3054
+#![cfg_attr(feature = "cargo-clippy", allow(redundant_field_names))]
```

Comment:
This is currently nessessary due to https://github.com/rust-lang-nursery/rust-clippy/issues/3054

Reply:
This alredy got fixed.